### PR TITLE
Join sig and mtp circuit

### PR DESCRIPTION
**Pull request by ilya-korotya**
Count of constraints for joined and separated circuits:
```bash
circom --r1cs circuits/credentialAtomicQuerySigMTP.circom
non-linear constraints: 69726

circom --r1cs circuits/credentialAtomicQuerySigV2.circom
non-linear constraints: 57184

circom --r1cs circuits/credentialAtomicQueryMTPV2.circom
non-linear constraints: 40372
```

I developed the new `credentialAtomicQuerySigMTP` circuit. This circuit combines verification SigV2 and MTPV2 circuits. Users can choose verification with `proofType` public input. 0 - for SigV2, 1 - for MTPV2.

### CODE REVIEW

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    // Verify issuerClaim schema
+    verifyCredentialSchema()(1, issuerClaim, claimSchema);
+
+    // verify issuerClaim expiration time
+    verifyExpirationTime()(issuerClaim, timestamp);
+    
+    signal isSig;
+    signal isMTP;
+    isSig <-- proofType==0 ? 1 : 0;
+    isMTP <-- proofType==1 ? 1 : 0;
```
Comment (AndriianChestnykh) [inline circuits/offchain/credentialAtomicQueryV3OffChain.circom:94]: May be this works fine.
However <-- is not really a constraint.
The following should work for sure and prevent from enabling both MTP and SIG at a time
```
isSig  <== 1 - proofType;
isMTP <== proofType;
isSig * isMTP === 0;
```

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        80551937543569765027552589160822318028
+    );
+
+    signal tmpAuthState;
+    tmpAuthState <== getIdenState()(
+        issuerAuthClaimsTreeRoot,
+        issuerAuthRevTreeRoot,
+        issuerAuthRootsTreeRoot
+    );
+    issuerAuthState <-- enabled ==1 ? tmpAuthState : 0;
```
Comment (OBrezhniev) [inline circuits/offchain/credentialAtomicQueryV3OffChain.circom:218]: Same problem with <--, change to similar code as proposed by Andriian.

**Review by OBrezhniev — CHANGES_REQUESTED** [review]

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    signal input claimPathKey; // hash of path in merklized json-ld document
+    signal input claimPathValue; // value in this path in merklized json-ld document
+
+    signal input slotIndex;
+    signal input operator;
+    signal input value[valueArraySize];
+
+    signal input issuerClaim[8];
+
+    // Diff for MTP proof
```
[the remark this answers is not in the document]
Comment (ilya-korotya) [reply circuits/offchain/credentialAtomicQueryV3OffChain.circom]: done

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom (hunk 1 above)
Comment (ilya-korotya) [reply circuits/offchain/credentialAtomicQueryV3OffChain.circom]: done

File: circuits/offchain/credentialAtomicQueryV3OffChain.circom (hunk 2 above)
Comment (ilya-korotya) [reply circuits/offchain/credentialAtomicQueryV3OffChain.circom]: done

**Review by OBrezhniev — APPROVED** [review]

**Review by AndriianChestnykh — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**